src/examples/regression_test.json

```diff
@@ -140,24 +140,24 @@
     "digest": "Group Primitive",
     "methods": {
       "add": {
-        "rows": 6,
-        "digest": "ef763443bed39d397cb02e683ea9a877"
+        "rows": 33,
+        "digest": "0cc19b3dc379f8706aedfd731f08145f"
       },
       "sub": {
-        "rows": 6,
-        "digest": "329c0221e60d62aaf26ee1631f3f3c5a"
+        "rows": 34,
+        "digest": "ff55b39b569856e1c2025cae39d19e38"
       },
       "scale": {
-        "rows": 106,
-        "digest": "1296dd40b232fb0bc4cd908d47fd88a9"
+        "rows": 114,
+        "digest": "17d1f373892d2c437dc9ce340e17e3d6"
       },
       "equals": {
-        "rows": 27,
-        "digest": "adbcdc756d41853e24612e50b99be285"
+        "rows": 42,
+        "digest": "8d5ab248aa4602e6f77a8ab1015615df"
```

Comment:
can you explain this blowup in `Group.equals()` constraints @Trivo25?

Reply:
is it just caused by the added constraints in `check()`?

Reply:
The `Group Primitive.equals` check bundles a couple additional equality checks than just a simple `equals()`, it does the following:

```ts
    let g1 = Provable.witness(Group, () => Group.generator);
    let g2 = Provable.witness(Group, () => Group.generator);
    g1.equals(g2).assertTrue();
    g1.equals(g2).assertFalse();
    g1.equals(g2).assertEquals(true);
    g1.equals(g2).assertEquals(false);
```

So its calling `equals` multiple times, which blows up the constraints a bit more. The single `equals` constraints aren't that bad, just inflated by the batch. The additional constraints come from `zero`-related checks (such as `.check`) as well as the changes to the `add` method

Reply:
yeah I understand that it has more, but the only thing that changed in that circuit should be `check()` and `equals()`, so I was wondering what explains the increase in circuit size compared to the previous version

Reply:
yea, its the change in `check`